nsq/async.py

```diff
@@ -277,7 +277,7 @@ def _read_body(self, data):
             self.trigger(event.DATA, conn=self, data=data)
         except Exception:
             logger.exception('uncaught exception in data event')
-        self.io_loop.add_callback(self._start_read)
```

Comment:
I have several spots where i use several independent reader's on the same IOLoop so yielding is important to me. It's also important because calling things directly can lead to infinite call stacks. I think that makes me :thumbsdown: for this change.